pkg/apis/internal.admin.acorn.io/v1/baseresources.go

```diff
@@ -75,43 +77,52 @@ func (current *BaseResources) Fits(incoming BaseResources) error {
 		}
 	}
 
-	// Check if any of the quantity resources are exceeded
-	for _, r := range []struct {
-		resource          string
-		current, incoming resource.Quantity
-	}{
-		{"VolumeStorage", current.VolumeStorage, incoming.VolumeStorage},
-		{"Memory", current.Memory, incoming.Memory},
-		{"Cpu", current.CPU, incoming.CPU},
-	} {
-		if !FitsQuantity(r.current, r.incoming) {
-			exceededResources = append(exceededResources, r.resource)
-		}
+	if len(exceededResources) != 0 {
+		errs = append(errs, fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", ")))
+	}
+
+	if err := current.ComputeClasses.Fits(incoming.ComputeClasses); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := current.VolumeClasses.Fits(incoming.VolumeClasses); err != nil {
+		errs = append(errs, err)
 	}
 
 	// Build an aggregated error message for the exceeded resources
-	if len(exceededResources) > 0 {
-		return fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", "))
+	if len(errs) != 0 {
+		return errors.Join(errs...)
```

Comment:
`errors.Join` does the right thing if `errs` is empty. You can unconditionally return its result.